src/main/scala/com/typesafe/sbt/packager/graalvm/GraalVMPlugin.scala

```diff
@@ -0,0 +1,50 @@
+package com.typesafe.sbt.packager.graalvm
+
+import sbt._
+import sbt.Keys._
+import java.nio.charset.Charset
+
+import com.typesafe.sbt.packager.SettingsHelper
+import com.typesafe.sbt.packager.Keys._
+import com.typesafe.sbt.packager.linux._
+import com.typesafe.sbt.packager.Compat._
+import com.typesafe.sbt.packager.validation._
+
+/**
+  * Plugin to compile ahead-of-time native executables.
+  *
+  * @example Enable the plugin in the `build.sbt`
+  * {{{
+  *    enablePlugins(GraalVMPlugin)
+  * }}}
+  */
+object GraalVMPlugin extends AutoPlugin {
+
+  object autoImport extends GraalVMKeys {
+    val GraalVM: Configuration = config("graalvm")
+  }
+
+  import autoImport._
+
+  override def projectConfigurations: Seq[Configuration] = Seq(GraalVM)
+
+  override lazy val projectSettings = Seq(
+    target in GraalVM := target.value / "graalvm",
+    sourceDirectory in GraalVM := sourceDirectory.value,
+    name in GraalVM := name.value,
+    nativeImageOptions in GraalVM := Seq(s"-H:Name=${(name in GraalVM).value}"),
+    packageBin in GraalVM := {
+      (target in GraalVM).value.mkdirs()
+      val className = (mainClass in Compile).value.getOrElse(sys.error("Could not find a main class."))
+      val classpath = Seq((packageBin in Compile).value) ++ (dependencyClasspath in Compile).value.map(_.data)
```

Comment:
Depending on packageBin and dependencyClasspath in compile is probably fine. If users want to build graal images in test scope we'll make this more flexible.